### PR TITLE
[14.0][MIG] website_apps_store: Migration to 14.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT30046

Depends on:
- [x] https://github.com/OCA/apps-store/pull/85

Main changes:
- Use reCaptcha v3 instead of reCaptcha v2 (inspired by the native Odoo website_form module).
- Improve tests.
- Now if there is a suspicious activity detected by Google reCaptcha during product download, you will be redirected to a page where the error message is displayed.

### CODE REVIEW

**Review by CarlosRoca13 — APPROVED** [review]

**Review by victoralmau — APPROVED** [review]
Code review OK.

File: website_apps_store/controllers/main.py [hunk 1]
```diff
@@ -186,35 +184,39 @@ def shop(self, page=0, category=None, search="", ppg=False, **post):
...
-        self, product_tmpl, product=False, google_captcha="", **kwargs
+        self, product_tmpl, product=False, recaptcha_token_response="", **kwargs
     ):
-        self.validate_recaptcha(google_captcha)
+        try:
+            # The except clause below should not let what has been done inside
+            # here be committed. It should not either roll back everything in
+            # this controller method. Instead, we use a savepoint to roll back
+            # what has been done inside the try clause.
+            request.params["recaptcha_token_response"] = recaptcha_token_response
```
Comment (yajo) [inline website_apps_store/controllers/main.py]: Please put this line before the `try`.

File: website_apps_store/static/src/scss/main.scss [hunk 2]
```diff
@@ -23,3 +23,12 @@
...
 }
+
+#product_detail[data-is-module="1"] {
+    #o-carousel-product .carousel-outer .carousel-inner img {
+        width: auto;
+    }
+    div.desc_rst {
+        margin-top: 50px;
+    }
+}
```
Comment (yajo) [inline website_apps_store/static/src/scss/main.scss]: If possible, add some bootstrap classes to those elements instead. It usually makes migrations more obvious when odoo uses BS 5.

See https://getbootstrap.com/docs/4.6/utilities/spacing/ and https://getbootstrap.com/docs/4.6/content/images/#responsive-images

**Review by yajo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Aborted merge due to latest comments.

**Pull request merged**